aten/src/ATen/core/op_registration/op_registration.cpp

```diff
@@ -165,4 +165,54 @@ RegisterOperators::~RegisterOperators() = default;
 RegisterOperators::RegisterOperators(RegisterOperators&&) noexcept = default;
 RegisterOperators& RegisterOperators::operator=(RegisterOperators&&) noexcept = default;
 
+
+CppFunction::CppFunction(KernelFunction func, std::unique_ptr<c10::FunctionSchema> schema)
+  : func_(std::move(func))
+  , schema_(std::move(schema))
+  {}
+
+Module::Module(const char* ns)
+  : ns_(ns)
+  {}
+
+Module::Module(Module&&) noexcept = default;
+Module& Module::operator=(Module&&) noexcept = default;
+
+
+Module&& Module::def(const char* schema) && {
```

Comment:
Using rvalue overloads and returning && is a way of pushing people towards using method chaining as in

    torch::import().def().def().def();

If you want to allow non-chained calls, i.e.

    auto m = torch::import().def();
    m.def();
    m.def();

then you might need to implement all those function twice, once with an rvalue overload that returns an rvalue (or a value) and once with an lvalue overload that returns a lvalue (this second overloads return can't be assigned to anything but it can still be used for chaining).

I guess it depends on what pybind does here and we should follow its api.

Reply:
pybind11 doesn't do chaining. Instead, the PYBIND11_MODULE macro brings an `m` into scope, and then people just call it like a normal method. There is no chaining support.

Reply:
wait, but in pybind, you can do `...def().def()` right?

For `m.def()` we need at least the lvalue overload here

Reply:
I think for now I won't add the lvalue overloads but we should talk about how "close to pybind11" we actually want the API to be next meeting.